pkg/services/app/add.go

```diff
@@ -292,7 +292,7 @@ func (a *App) getGitRemoteUrl(params AddParams) (string, error) {
 		return "", fmt.Errorf("remote config in %s does not have an url", params.Dir)
 	}
 
-	return sanitizeRepoUrl(urls[0]), nil
+	return utils.SanitizeRepoUrl(urls[0]), nil
```

Comment:
Seems like this whole function should either be replaced by your new `gogit.GetRemoteUrl` or be implemented in terms of it...

Reply:
I am feeling like we need to add some type safety to these URLs and handling git repository data in general. There are too many strings getting passed around. Probably outside the scope of this PR. I will create a follow up.

Reply:
I didn't mean the `SanitizeRepoUrl` function, I meant the function containing it `func (a *App) getGitRemoteUrl(params AddParams) (string, error)` (it basically duplicates `GetRemoteUrl`). I couldn't put a comment on the actual code above because it didn't change :-) I should have been more clear what I was referring to but that was the meaning of "whole function".

Reply:
I think we are in agreement. We rely on a URL being run through this function for lots of things to work. Same with cluster name. We can't just use any string, we need a `SanitizedURL` type or something.

Reply:
:+1: